finetune/base.py

```diff
@@ -956,16 +450,14 @@ def finetune_grid_search_cv(cls, Xs, Y, *, n_splits, test_size, config=None, eva
                 assert config == config_common
                 n_res += 1
                 sum_res += result
-            aggregated_results.append((config_common, sum_res/n_res))
+            aggregated_results.append((config_common, sum_res / n_res))
 
         if return_all:
             return aggregated_results
 
         return max(aggregated_results, key=lambda x: x[1])[0]
 
     def __del__(self):
-        try:
-            if self.sess is not None:
-                self.sess.close()
-        except AttributeError:
-            pass
+        for folder in self.tmp_dirs:
+            shutil.rmtree(folder)
+        self.tmp_dirs = []
```

Comment:
Pretty sure the reassignment isn't necessary because the object isn't going to be used if it's `__del__` method is called.

Reply:
Check if tmp_dirs exists before rmtree rather than adding after.